### ISSUE TITLE
Make Event a concrete struct instead of an interface

### ISSUE THREAD

**Issue by dolmen** [alevinval/sse#25]
1. The Event object are objects that only exit from the `sse` API: once they are out of the API, the consumer is free to do anything with it, as this will not affect us.
2. The definition is directly bound to the EventSource specification: the list of properties will not grow/shrink as long as the EventSource spec does not changes.

So I think that events should be exposed as concrete objects instead of interfaces. This removes one indirection layer.

I have a [branch that implements this change](/dolmen-go/sse/tree/concrete-Event) but it is based on master, not on #24.

**Comment by dolmen** [comment]
PS: I think that Event (whether it is a struct or an interface) has to change because it does not reflect the [MessageEvent](https://html.spec.whatwg.org/multipage/comms.html#messageevent) interface of the EventSource specification regarding the handling of IDs (ID vs [lastEventID](https://html.spec.whatwg.org/multipage/comms.html#dom-messageevent-lasteventid)).

**Comment by alevinval** [comment]
I see. I agree it needs to change, there is some work missing to fully mimic the MessageEvent interface. Also note the spec says the attributes should be readonly, interface looks nice for that, probably Data() should return string instead of []byte.

**Comment by dolmen** [comment]
My branch also makes Data as a string, but mostly because the spec says everything is UTF-8.
The readonly-ness of MessageEvent is irrelevant to us. As I wrote, our Event are just existing the API. If the consumer want to make them readonly he can still convert them to a private struct with a public interface. But we should not impose him the cost of the interface indirection while it is not necessary from the inside.
I will improve my branch once #24 is merged.

**Comment by alevinval** [comment]
I saw the commit, looks good. You convinced me. I just merged #24.

**Comment by alevinval** [comment]
Also, feel free to name it MessageEvent if you want.

**Comment by alevinval** [comment]
This pull request #28 has some improvements and fixes issue #26. However, I will bring some small conflicts on your work. I have no problem to keep it on hold until your work here is merged. If you don´t mind then I will merge it.
